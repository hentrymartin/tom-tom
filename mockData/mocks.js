export const incident = {
  id: 'europe_HD_DE_TTL6000',
  type: 1,
  point: {
    x: 13.072789,
    y: 52.388527,
  },
  from: 'Am Buchhorst (Arthur-Scheunert-Allee/L78)',
  to: 'Potsdam-Brauhausberg - Friedrich-Engels-Stra√üe (Heinrich-Mann-Allee/L78)',
  details: 'stationary traffic',
  delay: 232,
  magnitude: 2,
};

export const incidents = [];
