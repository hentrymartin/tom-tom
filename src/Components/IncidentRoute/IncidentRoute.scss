.route-affected {
  display: flex;
  align-items: center;
  &__from,
  &__to {
    font-size: 12px;
    max-width: 100px;
    padding: 10px;
    font-style: italic;
  }
  &__line {
    display: flex;
    flex: 1;
    border: 2px dashed #ffa7a7;
  }
}
