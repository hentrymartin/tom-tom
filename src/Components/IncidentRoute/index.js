import React from 'react';
import './IncidentRoute.scss';

const IncidentRoute = ({ from, to }) => (
  <div className="route-affected">
    <div className="route-affected__from">{from}</div>
    <div className="route-affected__line" />
    <div className="route-affected__to">{to}</div>
  </div>
);

IncidentRoute.defaultProps = {
  from: '',
  to: '',
};

export default IncidentRoute;
