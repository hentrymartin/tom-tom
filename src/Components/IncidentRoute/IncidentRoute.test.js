import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import IncidentRoute from './index';
import { incident } from '../../../mockData/mocks';

describe('Incident Route', () => {
  it('should render correctly', () => {
    const component = shallow(<IncidentRoute from={incident.from} to={incident.to} />);

    expect(component).toMatchSnapshot();
  });

  it('Check for from value', () => {
    const component = mount(
      <MemoryRouter>
        <IncidentRoute from={incident.from} to={incident.to} />
      </MemoryRouter>,
    );

    expect(component.find('.route-affected__from').text()).toBe(incident.from);
  });
});
