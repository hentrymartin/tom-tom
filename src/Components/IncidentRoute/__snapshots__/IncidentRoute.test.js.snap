// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Incident Route should render correctly 1`] = `ShallowWrapper {}`;

exports[`Incident should render correctly 1`] = `ShallowWrapper {}`;
